domain/secret/service/service.go

```diff
@@ -305,44 +301,43 @@ func (s *SecretService) UpdateUserSecret(ctx context.Context, uri *secrets.URI,
 		for k, v := range params.Data {
 			p.Data[k] = v
 		}
-	}
-
-	err := s.loadBackendInfo(ctx, true)
-	if err != nil {
-		return errors.Trace(err)
-	}
-	// loadBackendInfo will error is there's no active backend.
-	backend := s.backends[s.activeBackendID]
+		err := s.loadBackendInfo(ctx, true)
+		if err != nil {
+			return errors.Trace(err)
+		}
+		// loadBackendInfo will error is there's no active backend.
+		backend := s.backends[s.activeBackendID]
 
-	md, err := s.GetSecret(ctx, uri)
-	if err != nil {
-		// Check if the uri exists or not.
-		return errors.Trace(err)
-	}
-	revId, err := backend.SaveContent(ctx, uri, md.LatestRevision+1, secrets.NewSecretValue(params.Data))
-	if err != nil && !errors.Is(err, errors.NotSupported) {
-		return errors.Annotatef(err, "saving secret content to backend")
-	}
-	if err == nil {
-		defer func() {
-			if errOut != nil {
-				// If we failed to update the secret, we should delete the
-				// secret value from the backend for the new revision.
-				if err2 := backend.DeleteContent(ctx, revId); err2 != nil &&
-					!errors.Is(err2, errors.NotSupported) &&
-					!errors.Is(err2, secreterrors.SecretRevisionNotFound) {
-					s.logger.Warningf("failed to delete secret %q: %v", revId, err2)
+		md, err := s.GetSecret(ctx, uri)
```

Comment:
Can we add a TODO to use a bespoke "GetLatestRevision(ctx, uri) method instead of GetSecret()